packages/console/src/app/page.tsx

```diff
@@ -1,52 +1,67 @@
 'use client'
 
-import { useW3, Space } from '@storacha/ui-react'
-import { DidIcon } from '@/components/DidIcon'
-import Link from 'next/link'
+import { useState } from 'react'
+import { useW3 } from '@storacha/ui-react'
 import { SpacesNav } from './space/layout'
-import { H1, H2 } from '@/components/Text'
+import { H1 } from '@/components/Text'
 import SidebarLayout from '@/components/SidebarLayout'
-import { ReactNode } from 'react'
+import { SpacesTabNavigation } from '@/components/SpacesTabNavigation'
+import { SpacesList } from '@/components/SpacesList'
+import { UpgradePrompt } from '@/components/UpgradePrompt'
+import { usePrivateSpacesAccess } from '@/hooks/usePrivateSpacesAccess'
+import { useFeatureFlags } from '@/lib/featureFlags'
+import { useFilteredSpaces } from '@/hooks/useFilteredSpaces'
 
-export default function HomePage () {
+export default function HomePage() {
   return (
     <SidebarLayout>
       <SpacePage />
     </SidebarLayout>
   )
 }
 
-function SpacePage (): ReactNode {
+function SpacePage() {
+  const [activeTab, setActiveTab] = useState<'public' | 'private'>('public')
   const [{ spaces }] = useW3()
+  const { canAccessPrivateSpaces, shouldShowUpgradePrompt, planLoading } = usePrivateSpacesAccess()
+  const { canSeePrivateSpacesFeature } = useFeatureFlags()
+  const { publicSpaces, privateSpaces, hasHiddenPrivateSpaces } = useFilteredSpaces()
+
+  const shouldShowPrivateSpacesTab = canSeePrivateSpacesFeature
```

Comment:
nit: a single variable reassign just for the purpose of naming is very confusing. and, at this components level "canSeePrivateSpacesFeature" is the more correct variable name to describe intent for the whole page.

Reply:
also I hate to agree with @travis (as my arch nemisis), but actually usePrivateSpacesAccess has a "shouldShowPrivateSpacesTab" so this is really duplicative.